Migration.Tool.Extensions/DefaultMigrations/AssetMigration.cs

```diff
@@ -210,6 +221,62 @@ public async Task<FieldMigrationResult> MigrateValue(object? sourceValue, FieldM
                 }
             }
         }
+        else if (Kx13FormControls.UserControlForText.MediaSelectionControl.Equals(sourceFormControl, StringComparison.InvariantCultureIgnoreCase) && sourceValue is string sourceUrl)
+        {
+            if (!configuration.MigrateMediaToMediaLibrary)
+            {
+                // If we're migrating assets to content hub, unmatched URL can be stored as legacy media link
+                string folderPath = "/Legacy Media Links";
+                var folderGuid = GuidHelper.CreateFolderGuid(folderPath);
+                await contentFolderService.EnsureFolderStructure([new(folderGuid, folderGuid.ToString(), "Legacy Media Links", folderPath)]);
+                const int nameLength = 60;    // number of characters to take from the end of source url as representative name
+                string displayName = sourceUrl.Length >= nameLength ? sourceUrl[^nameLength..] : sourceUrl;
+                string name = await contentItemCodeNameProvider.Get(displayName);
+                var contentItemModel = new ContentItemSimplifiedModel
+                {
+                    CustomProperties = [],
+                    ContentItemGUID = GuidHelper.CreateContentItemGuid($"MediaSelectorLink|{valueKey}|{fieldName}"),
+                    ContentItemContentFolderGUID = folderGuid,
+                    IsSecured = null,
+                    ContentTypeName = AssetFacade.LegacyMediaLinkContentType.ClassName,
+                    Name = name,
+                    IsReusable = true,
+                    LanguageData = languageNames.Value.Select(lang =>
+                        new ContentItemLanguageData
+                        {
+                            LanguageName = lang,
+                            DisplayName = displayName,
+                            UserGuid = null,
+                            VersionStatus = VersionStatus.Published,
+                            ContentItemData = new Dictionary<string, object?>()
+                            {
+                                [AssetFacade.LegacyMediaLinkUrlField.Column!] = sourceUrl
+                            }
+                        }).ToList()
+                };
+                var importResult = await importer.ImportAsync(contentItemModel);
+                if (importResult is { Success: true })

```

Comment:
It is easy to overlook the start of the if condition when it's not separated from the previous code with blank line.